src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java

```diff
@@ -274,6 +288,10 @@ public final class OpenSearchSecurityPlugin extends OpenSearchSecuritySSLPlugin
     private volatile OpensearchDynamicSetting<Boolean> transportPassiveAuthSetting;
     private volatile PasswordHasher passwordHasher;
     private volatile DlsFlsBaseContext dlsFlsBaseContext;
+    private ResourceSharingIndexManagementRepository rmr;
+    private ResourceAccessHandler resourceAccessHandler;
+    private static final Map<String, ResourceProvider> RESOURCE_PROVIDERS = new HashMap<>();
```

Comment:
I usually advise against using `static` attributes in OpenSearch classes. So far, it is possible to instantiate several OpenSearch nodes and even clusters in parallel in one JVM. That's cool, as this allows to run several OpenSearch clusters in a JVM for testing. This is possible because OpenSearch managed mostly to avoid static fields and singletons. 

Having this static can lead to funny concurrency issues and side-effects.

Reply:
I have moved it to a new class called ResourcePluginInfo.